packages/components/src/TooltipTrigger.test.ts

```diff
@@ -0,0 +1,180 @@
+import { act, screen, within } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { button, h } from 'react-hyperscript-helpers';
+
+import { icon } from './icon';
+import { getPopupRoot } from './internal/PopupPortal';
+import { renderWithTheme } from './internal/test-utils';
+import { TooltipTrigger, TooltipTriggerProps } from './TooltipTrigger';
+
+describe('TooltipTrigger', () => {
+  const tooltipContent = 'This is a tooltip';
+
+  const renderTooltipTrigger = (
+    props: Partial<TooltipTriggerProps> = {},
+    childProps: Partial<JSX.IntrinsicElements['button']> = {}
+  ) => {
+    return renderWithTheme(
+      h(
+        TooltipTrigger,
+        {
+          content: tooltipContent,
+          ...props,
+        },
+        [button(childProps)]
+      )
+    );
+  };
+
+  it('shows tooltip when hovering on child', async () => {
+    // Arrange
+    const user = userEvent.setup();
+
+    renderTooltipTrigger();
+
+    const btn = screen.getByRole('button');
+
+    // Act
+    await user.hover(btn);
+    const isShownAfterHover = !!within(getPopupRoot()).queryByText(tooltipContent);
+
+    await user.unhover(btn);
+    const isShownAfterUnhover = !!within(getPopupRoot()).queryByText(tooltipContent);
+
+    // Assert
+    expect(isShownAfterHover).toBe(true);
+    expect(isShownAfterUnhover).toBe(false);
+  });
+
+  it('shows tooltip when focusing on child', () => {
+    // Arrange
+    renderTooltipTrigger();
+
+    const btn = screen.getByRole('button');
+
+    // Act
+    act(() => {
+      btn.focus();
+    });
+    const isShownAfterFocus = !!within(getPopupRoot()).queryByText(tooltipContent);
+
+    act(() => {
+      btn.blur();
+    });
+    const isShownAfterBlur = !!within(getPopupRoot()).queryByText(tooltipContent);
+
+    // Assert
+    expect(isShownAfterFocus).toBe(true);
+    expect(isShownAfterBlur).toBe(false);
+  });
+
+  it('calls child event handlers when hovering/unhovering', async () => {
+    // Arrange
+    const user = userEvent.setup();
+
+    const onMouseEnter = jest.fn();
+    const onMouseLeave = jest.fn();
+    renderTooltipTrigger({}, { onMouseEnter, onMouseLeave });
+
+    const btn = screen.getByRole('button');
+
+    // Act
+    await user.hover(btn);
+    await user.unhover(btn);
+
+    // Assert
+    expect(onMouseEnter).toHaveBeenCalled();
+    expect(onMouseLeave).toHaveBeenCalled();
+  });
+
+  it('calls child event handlers when focusing/blurring', () => {
+    // Arrange
+    const onFocus = jest.fn();
+    const onBlur = jest.fn();
+    renderTooltipTrigger({}, { onFocus, onBlur });
+
+    const btn = screen.getByRole('button');
+
+    // Act
+    act(() => {
+      btn.focus();
+    });
+
+    act(() => {
+      btn.blur();
+    });
+
+    // Assert
+    expect(onFocus).toHaveBeenCalled();
+    expect(onBlur).toHaveBeenCalled();
+  });
+
+  it('allows delaying showing the tooltip', async () => {
+    // Arrange
+    jest.useFakeTimers();
```

Comment:
I thought that maybe we implemented a wrapper for this... withJestFakeTImers maybe?

withJestFakeTimers(() => {  /* do test stuff */ })

implementation would be:

export const withJestFakeTimers(testFn: () => void): void {
  // setup timers
  testfn();
  // clearnup timers
  }

Reply:
We have not, but we could.

Reply:
Actually, I'd rather do this later. I'd want to put that helper in the test-utils package and before adding to test-utils, I want to update that package so that it's TypeScript.

Reply:
Then again... already added a test-utils module to this package, so I can add this here and then move it later.

Reply:
Added in https://github.com/DataBiosphere/terra-ui/pull/4245/commits/d4365d92f978c907dc9c76ee23b50001c341acd7.

Reply:
I was gonna ask why we couldn't move it to test-utils in the shared package... but you reminded me that that one is not like the rest, and still wants to be a "full package" when it grows up... So I am cool with that move being done after that love is given to make it "like the others"